docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,576 @@
+# Signal Generation
+
+An overview of the signal generation functionality available in this package, along with information about Tektronix signal generators.
+
+______________________________________________________________________
+
+## Function Generation
+
+A function is a limited set of common waveforms that are provided by default through the instrument.
+The simplicity of these waveforms allows for output parameters like waveform length and sample rate to be
+abstracted away. Frequency replaces these in order to provide signals which are easy to quantify and manipulate.
+
+Arbitrary Function Generators ({term}`AFG`s) utilize a phase increment process and a data lookup table to provide variable frequencies.
+The phase increment calculated is dependent on the waveform length and frequency requested. This has
+a side effect where the phase increment can be larger than one index in the lookup table. Functions bypass this
+issue by being simplistic enough that waveform length reduction doesn't have a detrimental effect on the end output.
+
+Arbitrary Waveform Generators ({term}`AWG`s) enforce one cycle per sample, allowing the output to be the same shape regardless of clock rate.
+The number of samples that occurs during a second is referred to as a sample per second (S/s), a unit which determines the frequency of the waveform.
+With low frequency functions, AWGs are functionally identical to AFGs, besides offering more constrained amplitudes and offsets.
+
+______________________________________________________________________
+
+### Class Structure
+
+```{mermaid}
+classDiagram
+    direction LR
+
+    SignalGeneratorMixin <|-- Tekscope
+    SignalGeneratorMixin <|-- AFG
+    SignalGeneratorMixin <|-- AWG
+
+```
+
+The `SignalGenerator` class is responsible for most waveform generators, including the `AWG` and `AFG`.
```

Comment:
Here and throughout the document you could link to the API if you want to by using 
```suggestion
The {py:obj}`tm_devices.drivers.pi.signal_generators.signal_generator.SignalGenerator` class is responsible for most waveform generators, including the `AWG` and `AFG`.
```

Reply:
Also, you don't reference the glossary here, but you do further in the paragraph. We should either always reference it, or only reference it once per term per page.

Reply:
I re reference the glossary under each new header section